app/models/host.rb

```diff
@@ -1859,6 +1859,10 @@ def read_only_storages
     storages.where(:host_storages => {:read_only => true})
   end
 
+  def available_storages
+    writable_storages.reject(&:maintenance)
```

Comment:
Yeah I wonder if this logic (what storage can be provisioned to) belongs in provisioning instead of the `Host` model?  It seems like the `allowed_storages` method is the right place for this type of logic.

@Fryguy actually were you thinking of `allowed_hosts` and `allowed_clusters`?  I don't see any `available_hosts` or `available_clusters` methods but there are `allowed_hosts` and `allowed_clusters` in `MiqRequestWorkflow`